### PR TITLE
KML admin interface

### PR DESCRIPTION
**Pull request by pfanguin**
Using a custom `bucketTimestampIndex`index to `query` instead of `scan`. The service is also adding a new `bucket`attribute, because dynamodb may only filter on _RangeIndex_ (in this case `timestamp`).

```
{
    u 'Table': {
        u 'GlobalSecondaryIndexes': [{
            u 'IndexSizeBytes': 6489026, u 'IndexName': u 'fileIdIndex', u 'Projection': {
                u 'ProjectionType': u 'ALL'
            }, u 'IndexStatus': u 'ACTIVE', u 'ProvisionedThroughput': {
                u 'NumberOfDecreasesToday': 0, u 'WriteCapacityUnits': 4, u 'LastIncreaseDateTime': 1437681114.25, u 'ReadCapacityUnits': 2, u 'LastDecreaseDateTime': 1437688256.433
            }, u 'KeySchema': [{
                u 'KeyType': u 'HASH', u 'AttributeName': u 'fileId'
            }, {
                u 'KeyType': u 'RANGE', u 'AttributeName': u 'timestamp'
            }], u 'IndexArn': u 'arn:aws:dynamodb:eu-west-1:443130343732:table/geoadmin-file-storage/index/fileIdIndex', u 'ItemCount': 63158
        }, {
            u 'IndexSizeBytes': 4939136, u 'IndexName': u 'bucketTimestampIndex', u 'Projection': {
                u 'ProjectionType': u 'ALL'
            }, u 'IndexStatus': u 'ACTIVE', u 'ProvisionedThroughput': {
                u 'NumberOfDecreasesToday': 0, u 'WriteCapacityUnits': 4, u 'ReadCapacityUnits': 2, u 'LastDecreaseDateTime': 1437688295.868
            }, u 'KeySchema': [{
                u 'KeyType': u 'HASH', u 'AttributeName': u 'bucket'
            }, {
                u 'KeyType': u 'RANGE', u 'AttributeName': u 'timestamp'
            }], u 'IndexArn': u 'arn:aws:dynamodb:eu-west-1:443130343732:table/geoadmin-file-storage/index/bucketTimestampIndex', u 'ItemCount': 44924
        }], u 'AttributeDefinitions': [{
            u 'AttributeName': u 'adminId', u 'AttributeType': u 'S'
        }, {
            u 'AttributeName': u 'bucket', u 'AttributeType': u 'S'
        }, {
            u 'AttributeName': u 'fileId', u 'AttributeType': u 'S'
        }, {
            u 'AttributeName': u 'timestamp', u 'AttributeType': u 'S'
        }], u 'TableArn': u 'arn:aws:dynamodb:eu-west-1:443130343732:table/geoadmin-file-storage', u 'ProvisionedThroughput': {
            u 'NumberOfDecreasesToday': 0, u 'WriteCapacityUnits': 5, u 'LastIncreaseDateTime': 1437693125.438, u 'ReadCapacityUnits': 10, u 'LastDecreaseDateTime': 1437693132.669
        }, u 'TableSizeBytes': 6489026, u 'TableName': u 'geoadmin-file-storage', u 'TableStatus': u 'ACTIVE', u 'KeySchema': [{
            u 'KeyType': u 'HASH', u 'AttributeName': u 'adminId'
        }], u 'ItemCount': 63158, u 'CreationDateTime': 1431459708.399
    }
}
```

Files `htpasswd` and `htgroup`are managed by `Puppet`

### CODE REVIEW

**Comment by pfanguin** [comment]
Test link : https://mf-chsdi3.dev.bgdi.ch/fap_kml_admin/admin/kml

**Comment by pfanguin** [comment]
jenkins test

**Comment by cedricmoullet** [comment]
@pfanguin what is the status of this ? can it be merged ?

**Comment by pfanguin** [comment]
Now, I think it's ready !

**Comment by cedricmoullet** [comment]
@gjn @loicgasser could you please review ?

File: apache/wsgi.conf.in [hunk 1]
```diff
@@ -115,3 +115,18 @@ AliasMatch ^${apache_entry_path}/kml/(map\.geo\.admin\.ch_KML_\d+\.kml)$ ${print
...
+RewriteCond %{HTTP:X-Forwarded-Proto} !https
+RewriteRule ^${apache_entry_path}/admin         https://%{HTTP_HOST}%{REQUEST_URI} [R,L]
+
+<LocationMatch ^${apache_entry_path}/admin/kml>
+AuthType Basic
+AuthName "Your IWI credential"
+# Optional line:
+AuthBasicProvider file
+AuthUserFile "/var/www/vhosts/mf-chsdi3/private/htpasswd"
+AuthGroupFile "/var/www/vhosts/mf-chsdi3/private/htgroup"
```
Comment (gjn) [inline apache/wsgi.conf.in]: These files are managed by puppet?

File: apache/wsgi.conf.in [hunk 2]
```diff
@@ -115,3 +115,18 @@ AliasMatch ^${apache_entry_path}/kml/(map\.geo\.admin\.ch_KML_\d+\.kml)$ ${print
 
 # Zadara downloads
 AliasMatch ^${apache_entry_path}/downloads/ch.(.*)$ ${vars:zadara_dir}ch.$1
+
+# KML admin
+RewriteCond %{HTTP:X-Forwarded-Proto} !https
+RewriteRule ^${apache_entry_path}/admin         https://%{HTTP_HOST}%{REQUEST_URI} [R,L]
+
+<LocationMatch ^${apache_entry_path}/admin/kml>
```
Comment (gjn) [inline apache/wsgi.conf.in]: Please use indent inside location match tag

File: chsdi/views/files.py [hunk 3]
```diff
@@ -21,14 +21,15 @@
...
+def _add_item(id, file_id=False, bucketname='public.geo.admin.ch'):
     table = get_dynamodb_table(table_name='geoadmin-file-storage')
     try:
         table.put_item(
             data={
                 'adminId': id,
                 'fileId': file_id,
-                'timestamp': time.strftime('%Y-%m-%d %X', time.localtime())
+                'timestamp': time.strftime('%Y-%m-%d %X', time.localtime()),
+                'bucket': bucketname
```
Comment (gjn) [inline chsdi/views/files.py]: Why was the bucketname added?

Comment (gjn) [inline chsdi/views/files.py]: Understood now. Please ignore.

**Comment by gjn** [comment]
LGTM. Would be glad if @loicgasser could also give a quick look as it involved dynamodb/files service.

**Pull request merged**